<?xml version="1.0" encoding="UTF-8"?><testrun name="CalTest" project="tdd-example" tests="1" started="1" failures="0" errors="1" ignored="0">
  <testsuite name="com.example.CalTest" time="0.002">
    <testcase name="cal_add" classname="com.example.CalTest" time="0.002">
      <error>java.lang.StringIndexOutOfBoundsException: Range [2, -1) out of bounds for length 8&#13;
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:55)&#13;
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:52)&#13;
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)&#13;
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)&#13;
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)&#13;
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex(Preconditions.java:112)&#13;
	at java.base/jdk.internal.util.Preconditions.checkFromToIndex(Preconditions.java:349)&#13;
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4602)&#13;
	at java.base/java.lang.String.substring(String.java:2715)&#13;
	at com.example.Cal.add(Cal.java:20)&#13;
	at com.example.CalTest.cal_add(CalTest.java:15)&#13;
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)&#13;
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)&#13;
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)&#13;
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)&#13;
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)&#13;
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)&#13;
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)&#13;
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)&#13;
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)&#13;
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)&#13;
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)&#13;
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)&#13;
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)&#13;
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)&#13;
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)&#13;
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)&#13;
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)&#13;
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)&#13;
&#13;
</error>
    </testcase>
  </testsuite>
</testrun>
